test/integration-tests/IntegrationTests.Helpers/TestHelper.cs

```diff
@@ -56,6 +59,36 @@ public Process StartSample(int traceAgentPort, string arguments, string packageV
                 processToProfile: executable);
         }
 
+        public ProcessResult RunSampleAndWaitForExit(int traceAgentPort, int? statsdPort = null, string arguments = null, string packageVersion = "", string framework = "", int aspNetCorePort = 5000)
+        {
+            var process = StartSample(traceAgentPort, arguments, packageVersion, aspNetCorePort: aspNetCorePort, statsdPort: statsdPort, framework: framework);
+
+            using var helper = new ProcessHelper(process);
+
+            process.WaitForExit(DefaultProcessTimeoutMs);
+            helper.Drain();
+            var exitCode = process.ExitCode;
```

Comment:
See: https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.waitforexit?view=net-5.0#System_Diagnostics_Process_WaitForExit_System_Int32_

> Returns true if the associated process has exited; otherwise, false.

I think that `process.WaitForExit(int)` does not terminate the process. I guess that we should kill it. Otherwise the process may leak and probably `process.ExitCode` will throw an exception.

Reply:
true, fixed.